src/patternfly/components/List/list.scss

```diff
@@ -43,4 +75,11 @@
       }
     }
   }
+
+  &.pf-m-bordered {
+    > li:not(:last-child) {
```

Comment:
One thing about this is that it will create a fairly high specificity for this rule. There are a few ways you can improve that.

```
> li {
  border-bottom...
> li:last-child {
  border-bottom: 0;
```

Or you can rewrite that the same way with the universal selector, since the only valid child of a list is the `li` element, and the universal selector has 0 specificity.

```
> {
  border-bottom...
> :last-child {
  border-bottom: 0;
```

You can also write this using a [lobotomized owl selector](https://css-tricks.com/lobotomized-owls/), though it will change the way borders appear with nested lists.

```
> * + * {
  border-top...
```

Reply:
+1 for universal, direct descendant 

```
> * {
  border-bottom...
> :last-child {
  border-bottom: 0;
```